packages/grafana-data/src/types/annotations.ts

```diff
@@ -8,17 +8,23 @@ import { DataFrame } from './dataFrame';
  * This JSON object is stored in the dashboard json model.
  */
 export interface AnnotationQuery<TQuery extends DataQuery = DataQuery> {
-  datasource: string;
+  datasource?: string | null;
   enable: boolean;
   name: string;
   iconColor: string;
   hide?: boolean;
+  builtIn?: number;
+  type?: string;
+  snapshotData?: any;
 
   // Standard datasource query
   target?: TQuery;
 
   // Convert a dataframe to an AnnotationEvent
   mappings?: AnnotationEventMappings;
+
+  // Sadly plugins can set any propery directly on the main object
+  [str: string]: any;
```

Comment:
```suggestion
  [key: string]: any;
```
Nit: More descriptive name.